src/lib/components/chat/Messages.svelte

```diff
@@ -221,6 +222,19 @@
 			scrollToBottom();
 		}, 100);
 	};
+
+	const deleteMessagePair = async (messageId) => {
+		history.messages[messageId].deleted = true;
+		history.messages[history.messages[messageId].childrenIds[0]].deleted = true;
+
+		const responseId = history.messages[messageId].childrenIds[0];
+		if (history.messages[responseId].childrenIds.length === 0) {
```

Comment:
this part should probably use `message.done` instead of checking for number of children.

Reply:
The logic for this is that we should not cancel the chat completion if the deleted message isn't the most recently sent.

example:
User: Prompt1
Llama: Response1
User: Prompt2
Llama: Response2
User: Prompt3
Llama: generating...

In the above scenario, if the user deletes Prompt2 we still want to continue to generate the Response to Prompt3

Reply:
Wouldn't the following do the same job? 

```js
const childrenIds = history.messages[messageId].childrenIds;
if (!history.messages[childrenIds.at(-1)].done) {
	await cancelChatCompletion(localStorage.token, currentRequestId);
}	
```
Also, the second param should be `currentRequestId` not `chatId` 😅 it might be better to remove this part of code entirely, and we can introduce this feature in a separate PR.

Reply:
I agree, sounds good, thanks for pointing that out!